public/app/features/explore/state/utils.ts

```diff
@@ -86,11 +87,12 @@ export const createEmptyQueryResponse = (): ExplorePanelData => ({
 
 export async function loadAndInitDatasource(
   orgId: number,
-  datasourceUid?: string
```

Comment:
This still gets called from places where the URL might send in a name as a string, so we allow both types here. Basically we should let this be equivalent to whatever `datasourceSrv.get` accepts.